### PR TITLE
Clarify `quarkus.config.locations` and multiple profiles documentation

### LINKED ISSUE #35410: Regression 3.3.0 config quarkus.config.locations with vault
### Describe the bug

When switching from `3.2.4`  to `3.3.0` , loading properties from vault does not work anymore when using `quarkus.config.locations`. It may apply to other config source extensions. Not sure at this point.

### Expected behavior

property should be loaded from vault, as in `3.2.4`

### Actual behavior

the property is not loaded.

### How to Reproduce?

create an application with quarkus `3.3.0` and vault extension `3.1.0` .
start a vault like in the quickstart:
```
docker run --rm --cap-add=IPC_LOCK -e VAULT_ADDR=http://localhost:8200 -p 8200:8200 -d --name=dev-vault vault
docker logs dev-vault
...
# get Root Token: from the logs
docker exec -it dev-vault sh
...
# inside shell
export VAULT_TOKEN=...
vault kv put secret/myapps/vault-quickstart/config a-private-key=123456
```

in a directory somewhere outside the app:
- create a `config` directory
- create 2 files: `application.properties` and `base-application.properties`

in `application.properties`, add:
```
quarkus.config.locations=./config/base-application.properties
```

in `base-application.properties`, add:
```
config_ordinal=255

#quarkus.http.port=8090
quarkus.vault.kv-secret-engine-version=2

quarkus.vault.url=http://localhost:8200
quarkus.vault.tls.skip-verify=true
quarkus.vault.secret-config-kv-path.quickstart=myapps/vault-quickstart/config
quarkus.vault.authentication.client-token=<root token>
foo=${quickstart.a-private-key}

quickstart.a-private-key=xxx

quarkus.log.category."io.quarkus.vault".level=DEBUG

```

in `GreetingResource`:
``` 
    @ConfigProperty(name = "foo")
    String foo;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    @Path("/prop")
    public String helloProp() {
        return foo;
    }
``` 

start the application. you get:
```
$ java -Dquarkus.http.port=8090 -jar ../target/quarkus-app/quarkus-run.jar
Failed to load config value of type class java.lang.String for: foo
```

it works if:
- you switch to quarkus `3.2.4`  and vault `3.0.0`  
- you add property `quickstart.a-private-key=xxx` to `base-application.properties`
- you move the content of `basde-application.properties` into `application.properties`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.3.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #35410

### CODE REVIEW

File: docs/src/main/asciidoc/config-reference.adoc [hunk 1]
```diff
@@ -147,6 +147,25 @@ quarkus.http.port=9090 <2>
 TIP: It works in the exact same way as Quarkus Application configuration file `application.properties`. Recommendation
 is to use Quarkus `application.properties`.
 
+=== Locations
+
+Additionally, to the default config locationa, Quarkus provides a way to scan additional locations for configuration
```
Comment (gsmet) [inline docs/src/main/asciidoc/config-reference.adoc:152]:
```suggestion
Additionally to the default config locations, Quarkus provides a way to scan additional locations for configuration
```

File: docs/src/main/asciidoc/config-reference.adoc [hunk 2]
```diff
@@ -147,6 +147,25 @@ quarkus.http.port=9090 <2>
...
+
+- file or directory (`file:`)
+- classpath resource
+- jar resource (`jar:`)
+- http resource (`http:`)
+
+All loaded sources use the same ordinal of the source that found the `quarkus.config.locations` configuration
+property. For instance, if `quarkus.config.locations` is set as a system property, then all loaded sources have their
+ordinals set to `400` (system properties use `400` as their ordinal). The ordinal may be overridden directly in the
+resource by setting the `config_ordinal` property. Sources are sorted first by their ordinal, then by location order,
```
Comment (gsmet) [inline docs/src/main/asciidoc/config-reference.adoc:166]: Which `resource`? How do we set `config_ordinal` there?

File: docs/src/main/asciidoc/config-reference.adoc [hunk 3]
```diff
@@ -359,7 +378,8 @@ Properties in the profile aware file have priority over profile aware properties
 
 [WARNING]
 ====
-The profile aware file must be present in the exact same location as the main `application.properties` file.
+A profile aware file is only loaded if the unprofiled `application.properties` is also available in the same location.
+This is required to keep a  consistent loading order and pair all the resources together.
```
Comment (gsmet) [inline docs/src/main/asciidoc/config-reference.adoc:382]:
```suggestion
This is required to keep a consistent loading order and pair all the resources together.
```

File: docs/src/main/asciidoc/config-reference.adoc [hunk 4]
```diff
@@ -359,7 +378,8 @@ Properties in the profile aware file have priority over profile aware properties
 
 [WARNING]
 ====
-The profile aware file must be present in the exact same location as the main `application.properties` file.
+A profile aware file is only loaded if the unprofiled `application.properties` is also available in the same location.
```
Comment (gsmet) [inline docs/src/main/asciidoc/config-reference.adoc:381]: Wondering if we should add ``(respectively, `.yml` or `.yaml`)`` here? Not sure if it makes sense. I know we avoid having YAML content in this doc but I'm wondering if we should be clear that the extension needs to be exactly the same.

**Review by gsmet — CHANGES_REQUESTED** [review]
I spotted a few typos and added a few questions.

File: docs/src/main/asciidoc/config-reference.adoc (hunk 2 above)
Comment (radcortez) [reply docs/src/main/asciidoc/config-reference.adoc:166]: It's a specific thing coming from the MP Config spec. Let me clarify it.

File: docs/src/main/asciidoc/config-reference.adoc (hunk 4 above)
Comment (radcortez) [reply docs/src/main/asciidoc/config-reference.adoc:381]: We can add that. It should be ok.

**Review by gsmet — APPROVED** [review]

**Pull request merged**